system/default_ad_api_helpers/ad_api_adaptors/launch/rviz_adaptors.launch.xml

```diff
@@ -1,6 +1,14 @@
 <launch>
-  <node pkg="ad_api_adaptors" exec="initial_pose_adaptor" name="initial_pose_adaptor">
-    <remap from="~/input/goal" to="/planning/mission_planning/goal"/>
-    <remap from="~/input/waypoint" to="/planning/mission_planning/checkpoint"/>
-  </node>
+  <group>
+    <push-ros-namespace namespace="default_ad_api/helpers"/>
+    <node pkg="ad_api_adaptors" exec="initial_pose_adaptor" name="initial_pose_adaptor">
+      <param from="$(find-pkg-share ad_api_adaptors)/config/initial_pose.param.yaml"/>
+      <remap from="~/initialpose" to="/initialpose"/>
+      <remap from="~/fit_map_height" to="/localization/util/fit_map_height"/>
```

Comment:
I agree. :pray:

Reply:
This adapter exposes `initialpose` for rviz, but not `fit_map_height`. It's just for internal use.

Reply:
I'm sorry, I couldn't understand well. How can others know it's for internal use? It looks the same as `/initialpose`.

Reply:
Anything not documented in the AD API is an internal implementation. From the point of view of AD API, initialpose is also treated as an internal implementation. It's just connected to rviz and localization in the same way that other component node is connected to multiple components.

Reply:
Please let me organize my question.

In [this comment](https://github.com/autowarefoundation/autoware.universe/pull/1431#discussion_r958703883), you said as if `initialpose` and `fit_map_height` are different.
My question was how others can understand the difference.

And my proposal is, by not exposing `fit_map_height` here, people can understand it.
Since `fit_map_height` is not used in RViz, I feel it's reasonable.

Reply:
I cannot understand what your problem is. It's common for a launch file to list both the interfaces a node provides and the interfaces it depends on. For example, [ad_service_state_monitor.launch.xml](https://github.com/autowarefoundation/autoware.universe/blob/main/system/ad_service_state_monitor/launch/ad_service_state_monitor.launch.xml) exposes `output/autoware_state` and `output/autoware_engage`, and uses implementation detail topic such as `input/vector_map`, `input/modified_goal`. What you are saying is the same as removing these.

Reply:
I feel it's strange that `initial_pose_adaptor` has the interface `fit_map_height`.
Ideally, I think as follows:
- `initial_pose_adaptor` only has the interface `initialpose`.
- When it receives an initialpose, it calls the service like `/localization/initialize_pose`.
- During `/localization/initialize_pose`, the localization component internally calls `fit_map_height`.
- So, `fit_map_height` won't be an interface of `initial_pose_adaptor`.

But since it's not so a big problem, if you don't agree with me and others don't have any comments, the current implementation is acceptable.

Reply:
Do you consider this to be implementation independent? It's different and this adaptor is the **default implementation**. In other words, this is the internal implementation that connects rviz with the API.

Reply:
>During /localization/initialize_pose, the localization component internally calls fit_map_height.
So, fit_map_height won't be an interface of initial_pose_adaptor.

It cannot. The API must trust the pose of `localization/initialize_pose`. The adapter modifies the height because it knows it's from rviz and is unreliable.

Reply:
I'm not sure about the default implementation, but I feel the localization component itself ideally should support 3D pose initialization.

If you call `fit_map_height` from `initial_pose_adaptor`, the issue will be hidden.
I mean, AD API can support 3D pose initialization, but the localization component only supports 2D pose initialization + height adjustment. Is that the proper component interface? :thinking:

Reply:
API and localization component only supports 3D pose initialization. This adaptor makes 3D pose from 2D pose initialization + height adjustment. Localization uses `fit_map_height` only for GNSS pose.

Reply:
Ah, I see. So you mean the height adjustment is RViz specific.
Please let me think it again.

Reply:
Regarding the current implementation, I think it's acceptable, but feel a bit unnatural because the localization component has the interface `/localization/util/fit_map_height`.

So, let's proceed with the current implementation and revisit the interface in the future.
@isamu-takagi Is that okay for you?

Reply:
Alternatively, make `fit_map_height` a library and include it in each of GNSS (pose initializer) and RViz (API adaptor) node. It's a clean separation by design, but uses extra memory for the PCD map.

Reply:
Sounds good, Thanks.
Let's reconsider at a later stage if necessary.